src/dbus/dbus_object.hpp

```diff
@@ -67,44 +70,117 @@ class DBusObject
+     * @param[in]   aMethodName       The method name.
+     * @param[in]   aHandler          The method handler.
+     */
+    void RegisterGetPropertyHandler(const std::string &        aInterfaceName,
+                                    const std::string &        aMethodName,
+                                    const PropertyHandlerType &aHandler);
+
+    /**
+     * This method the set handler for a property.
+     *
+     * @param[in]   aInterfaceName    The interface name.
+     * @param[in]   aMethodName       The method name.
+     * @param[in]   aHandler          The method handler.
      */
-    void RegisterMethod(const std::string &aInterfaceName, const std::string &aMethodName, const HandlerType &aHandler);
+    void RegisterSetPropertyHandler(const std::string &        aInterfaceName,
+                                    const std::string &        aPropertyName,
+                                    const PropertyHandlerType &aHandler);
 
     /**
      * This method sends a signal.
      *
-     * @param[in]   aInterfaceName    The interface name
-     * @param[in]   aSignalName       The signal name
-     * @param[in]   aArgs             The tuple to be encoded into the signal
+     * @param[in]   aInterfaceName    The interface name.
+     * @param[in]   aSignalName       The signal name.
+     * @param[in]   aArgs             The tuple to be encoded into the signal.
+     *
+     * @retval OTBR_ERROR_NONE  Signal successfully sent.
+     * @retval OTBR_ERROR_DBUS  Failed to send the signal.
      */
     template <typename... FieldTypes>
-    void Signal(const std::string &              aInterfaceName,
-                const std::string &              aSignalName,
-                const std::tuple<FieldTypes...> &aArgs)
+    otbrError Signal(const std::string &              aInterfaceName,
+                     const std::string &              aSignalName,
+                     const std::tuple<FieldTypes...> &aArgs)
+    {
+        UniqueDBusMessage signalMsg = UniqueDBusMessage(
+            dbus_message_new_signal(mObjectPath.c_str(), aInterfaceName.c_str(), aSignalName.c_str()));
+        otbrError err = OTBR_ERROR_NONE;
+
+        VerifyOrExit(signalMsg != nullptr, err = OTBR_ERROR_DBUS);
+        VerifyOrExit(err = otbr::dbus::TupleToDBusMessage(*signalMsg, aArgs));
+
+        VerifyOrExit(dbus_connection_send(mConnection, signalMsg.get(), nullptr), err = OTBR_ERROR_DBUS);
+    exit:
+        return err;
+    }
+
+    /**
+     * This method sends a property changed signal.
+     *
+     * @param[in]   aInterfaceName    The interface name.
+     * @param[in]   aPropertyName     The property name.
+     * @param[in]   aValue            New value of the property.
+     *
+     * @retval OTBR_ERROR_NONE  Signal successfully sent.
+     * @retval OTBR_ERROR_DBUS  Failed to send the signal.
+     */
+    template <typename ValueType>
+    otbrError SignalPropertyChanged(const std::string &aInterfaceName,
```

Comment:
nit:  `SignalPropertyChanged` -> `SignalPropertiesChanged` to match the macro `DBUS_PROPERTIES_CHANGED_SIGNAL`?

Reply:
This function only support sending the signal message of one property's change.